crates/turbopack-core/src/chunk/chunking.rs

```diff
@@ -12,29 +12,59 @@ use regex::Regex;
 use tracing::Level;
 use turbo_tasks::{ReadRef, TryJoinIterExt, ValueToString, Vc};
 
-use super::{AsyncModuleInfo, Chunk, ChunkItem, ChunkType, ChunkingContext};
+use super::{
+    AsyncModuleInfo, Chunk, ChunkItem, ChunkItemsWithAsyncModuleInfo, ChunkType, ChunkingContext,
+    Chunks,
+};
 use crate::output::OutputAssets;
 
+#[turbo_tasks::value]
+struct ChunkItemInfo {
+    ty: Vc<Box<dyn ChunkType>>,
+    name: Vc<String>,
+    size: usize,
+}
+
+#[turbo_tasks::function]
+async fn chunk_item_info(
+    chunking_context: Vc<Box<dyn ChunkingContext>>,
+    chunk_item: Vc<Box<dyn ChunkItem>>,
+    async_info: Option<Vc<AsyncModuleInfo>>,
+) -> Result<Vc<ChunkItemInfo>> {
+    let asset_ident = chunk_item.asset_ident().to_string();
+    let ty = chunk_item.ty().resolve().await?;
+    let chunk_item_size = ty.chunk_item_size(chunking_context, chunk_item, async_info);
+    Ok(ChunkItemInfo {
+        ty,
+        size: *chunk_item_size.await?,
+        name: asset_ident.resolve().await?,
+    }
+    .cell())
+}
+
 /// Creates chunks based on heuristics for the passed `chunk_items`. Also
 /// attaches `referenced_output_assets` to the first chunk.
-#[tracing::instrument(level = Level::TRACE, skip_all)]
+#[turbo_tasks::function]
 pub async fn make_chunks(
```

Comment:
It's a turbo-tasks function now, so changes to chunk_item_size no longer invalidate the calling function